src/CameraSensor.cc

```diff
@@ -394,6 +394,12 @@ bool CameraSensor::Update(const std::chrono::steady_clock::duration &_now)
   // move the camera to the current pose
   this->dataPtr->camera->SetLocalPose(this->Pose());
 
+  if (this->dataPtr->infoPub.HasConnections())
```

Comment:
I moved the infoPub up here. Else, the camera_info wouldn't be published if the camera is a Triggered camera.